book/src/proposals/booting-from-snapshot.md

```diff
@@ -0,0 +1,223 @@
+# Booting From a Snapshot
+
+## Problem
+
+A validator booting from a snapshot should confirm that they are caught up to
+the correct cluster before voting. There are two major risks if they do not:
+
+### Wrong Cluster
+
+The validator may be fed a snapshot and gossip info for a different (though not
+necessarily malicious) cluster. If that cluster is at slot 100,000 while the
+cluster the validator intended to join is at 10,000, if the validator votes
+once on the wrong cluster, it is locked out of the correct cluster for 90,000
+slots. This is because slashing is based on slot numbers, so when the validator
+votes on the wrong cluster's slot 100,000, it can't make any vote on the
+correct cluster that locks it out of the correct cluster's slot 100,000. Due to
+this, it can't achieve `MAX_LOCKOUT` on the correct cluster until that goes
+past slot 100,000.
+
+### Voting Before Catching Up
+
+Any vote that a validator signs must have its lockout observed. If a validator
+has to reboot, the most secure way for it to check if its lockouts apply to the
+new set of forks is to see the votes on the ledger. If it can do this, it knows
+that those votes do not lock it out of voting on the current fork, so it just
+needs to wait for the lockouts of the votes not on the ledger to expire.
+However, if the validator votes before catching up, the votes will not go onto
+the ledger, so if the validator reboots, it will have to assume that the votes
+lock it out from voting again.
+
+
+## Snapshot Verification Overview
+
+A booting validator needs to have some set of trusted validators whose votes it
+will rely upon to ensure that it has caught up to the cluster and has a valid
+snapshot. With that set, the validator can follow the procedure detailed in the
+`Boot Procedure` section.
+
+For the sections below:
+
+Assume an arbitrary locktower state `L`, a snapshot root `S`, and a trusted
+bank slot `T` (defined in step 7-1 of the `Boot Procedure` section).
```

Comment:
We can get rid of this trusted bank slot `T` and the reference to it below, it's not actually used n the correctness reasoning